code/modules/mob/living/simple_animal/hostile/bees.dm

```diff
@@ -245,11 +252,26 @@
 /obj/item/queen_bee/attackby(obj/item/I, mob/user, params)
 	if(istype(I,/obj/item/weapon/reagent_containers/syringe))
 		var/obj/item/weapon/reagent_containers/syringe/S = I
-		var/datum/reagent/R = chemical_reagents_list[S.reagents.get_master_reagent_id()]
-		if(R)
-			queen.assign_reagent(R)
-			user.visible_message("<span class='warning'>[user] injects [src]'s genome with [R.name], mutating it's DNA!</span>","<span class='warning'>You inject [src]'s genome with [R.name], mutating it's DNA!</span>")
-			name = queen.name
+		if(S.reagents.has_reagent("royal_bee_jelly")) //checked twice, because I really don't want royal bee jelly to be duped
+			if(S.reagents.has_reagent("royal_bee_jelly",5))
```

Comment:
oh right you can't read that?

If it has the reagent, ANY AMOUNT, it avoids the else below, it's the else that adds the reagent to the bee.
so now we've avoided putting it in bees, fixing what I would consider too cheap/op, the SECOND check is that you have enough jelly to actually DO the transformation, which is 5u.